### DISCUSSION TITLE
Understanding the performance of a query

### DISCUSSION THREAD

**Discussion by rmkerr** [supabase-community/supabase-csharp#61] [Q&A] [answered]
I've written the following query:

```csharp
 var result = await _supabaseClient.From<KnownEndpointEntity>().Select("*").Where(x => x.id == id).Get();
 return result.Models.First();
```

I'm trying to understand if the filtering (where clause) is being done in the database in a query, or on the client. I've read the docs that I can find for both `supabase-csharp` and `postgrest-csharp` but have not been able to figure it out. Ideally if the filtering is being done in-client I'd like to find a way to better structure this filtering to happen on the database side.

I'll keep investigating and will update here if I figure it out myself.

**Comment by acupofjose** [comment, marked as answer]
Filtering happens database side - I’ve written the LINQ parser so that it translates the query expressions into the GET/POST/PATCH requests necessary to interact with `postgrest-csharp`.

Basically, everything that happens prior to the async request is server side:
```c#
var response = await postgrestClient
   .Table<T>()
   .Where(...) // Server Side
   .Where(...) // Server Side
   .Get();
``` 
 Everything that happens with `response.Models<T>()` is client side.

```c#
var models = response.Models<T>();

var filtered = models.Where(....) // Client Side
```

**Reply by rmkerr** [reply]
Very cool! Thanks for the quick response Joseph.

**Reply by rhuanbarros** [reply]
Hi, I implemented filtering in the frontend in the example. You can check the code [here](https://github.com/supabase-community/supabase-csharp/blob/master/Examples/BlazorWebAssemblySupabaseTemplate/Pages/Crud/CrudPage.razor.cs) and can test it [here](https://blazorwasmsupabasetemplate.web.app/crud). It's a simple implementation though.

If you want server side implementation, you could see [this example code from MudBlazor](https://mudblazor.com/components/table#server-side-filtering,-sorting-and-pagination), but you will need to implement the _ServerReload_ method to work with supabase api.
